File: typescript/src/singleton/singleton.test.ts
import { Logger } from "./singleton"

test("When execute newLogger, it should return the same instance of Logger every time", () => {
  const instance1 = Logger.newLogger()
  const instance2 = Logger.newLogger()

  expect(instance1).toBe(instance2)
  expect(instance1 === instance2).toBeTruthy()
})

File: go/singleton/singleton_test.go
package singleton

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestNewLogger(t *testing.T) {
	instance1 := NewLogger()
	instance2 := NewLogger()

	assert.Equal(t, instance1, instance2)
	assert.Equal(t, &instance1, &instance2)
}
